lib/src/component_declaration/component_base.dart

```diff
@@ -129,6 +145,10 @@ typedef TProps BuilderOnlyUiFactory<TProps extends UiProps>();
 abstract class UiComponent<TProps extends UiProps> extends react.Component implements DisposableManagerV7 {
   Disposable _disposableProxy;
 
+  bool propsAreEqual(TProps otherProps) {
```

Comment:
Do we have enough use-cases to:
- justify adding this as an API to over_react?
- know whether this should be the final API?

For example, there are some prop comparisons that exclude children, and there may also be others that perform deep equality on the children List.

Could we hold off until this is more solidified? I'd like to avoid committing to an API that may need to change soon.

---

Other comments:

This doesn't need to be an instance method, so could we make it a top-level function instead?

This needs a doc comment.